### ISSUE TITLE
Error: Mark cannot be in two anchor classes

### ISSUE THREAD

**Issue by hyvyys** [googlefonts/fontmake#727]
I have UFO sources with the following lines in the features:

```
    markClass uni0328 <anchor 0 0> @bottom;
    markClass uni0328 <anchor 0 0> @ogonek;
```

This is quite natural so that the ogonek combines where appropriate with the letters that have the specific ogonek anchor and combines in the center with other letters. This compiles fine in fontmake.

However as soon as I add some mark to ligature code, e.g.:

```
pos ligature c_a.liga
<anchor 262 0> mark @bottom
<anchor 262 500> mark @top
<anchor 286 0> mark @ogonek
ligComponent
<anchor 676 500> mark @top
<anchor 666 0> mark @bottom
<anchor 796 0> mark @ogonek;
```

I get an error:

```
fontmake: Error: In 'Gestural Sans.designspace': Generating fonts from Designspace failed: /FontFamily_Project/masters/FontFamily.ufo/features.fea:2605:1: Glyph uni0328 cannot be in both @bottom and @ogonek
```

**Comment by hyvyys** [comment]
And by the way, when I remove the bottom anchor from uni0328, I get another one:

```
fontmake: Error: In 'FontFamily.designspace': Generating fonts from Designspace failed: ((34, [33, 34]), 'list', '.LigatureAttach', 'LigatureArray', '.LigatureArray', 'MarkLigPos', '[0]', 'Lookup', '[4]', 'list', '.Lookup', 'LookupList', '.LookupList', 'GPOS', '.table', 'table_G_P_O_S_')
```

:(

**Comment by hyvyys** [comment]
I must add this happens when trying to build a variable TTF. Building static TTFs is possible without errors, however the mark-to-ligature attachment is not interpolated and all instances get the positions from the first master.

**Comment by anthrotype** [comment]
the error about "Glyph uni0328 cannot be in both @bottom and @ogonek" is correct, you can't use overlapping markClass definitions within the same lookup subtable.

The second error seem to indicate that the masters' feature files contain incompatible mark-to-liga definitions, which leads to varLib failing to interpolate them. Hard to tell exactly what caused the problem without looking at the actual sources.

Interpolating hand-written GPOS features when building static TTFs is not supported currently; only the auto-generated kern and mark/mkmk features will be interpolated (i.e. the data from kerning.plist and the anchors positions that are used respectively to generate those features). In your case, you opted for writing your own custom mark feature code, which means (by default) fontmake will skip generating the automatic feature based on the anchor names and positions.
There is a way to tell the MarkFeatureWriter to "append" the automatic features after the user-written ones, but I'm not sure that's what you're after.

**Comment by hyvyys** [comment]
Right, I forgot about subtables because FontLab generated the code. I used to play with all that when I still used FontForge. But let's leave that as attaching ogonek manually to random letter is science fiction territory. 

Moving back to mark-to-ligature anchors, you're totally right, I had some incompatibility and after fixing it I built the VF correctly. Thanks!

As for static fonts and auto-generated features, from what I gather, fontmake cannot process multiple anchors (defined as <anchor> in UFO) with the same name:

```
WARNING:ufo2ft.featureWriters.markFeatureWriter.MarkFeatureWriter:duplicate anchor 'bottom' in glyph 'p_t.liga'
```

and I have no idea how I would specify which anchor goes in which component. Since even the UFO spec doesn't specify this, I doubt this is implemented?
